### PR TITLE
chore: libp2p 0.34.1

### PR DESCRIPTION
**Pull request by hacdias**
Depends on 

- [x] https://github.com/ipfs/boxo/pull/611
- [x] https://github.com/libp2p/go-libp2p/pull/2810

### CODE REVIEW

File: core/node/libp2p/rcmgr_logging.go [hunk 1]
```diff
@@ -118,6 +118,20 @@ func (n *loggingResourceManager) OpenConnection(dir network.Direction, usefd boo
 	return connMgmtScope, err
 }
 
+func (n *loggingResourceManager) OpenConnectionNoIP(dir network.Direction, usefd bool, endpoint ma.Multiaddr) (network.ConnManagementScope, error) {
+	var connMgmtScope network.ConnManagementScope
+	var err error
+
+	if rcmgr, ok := n.delegate.(rcmgrOpenConnectionNoIPer); ok {
+		connMgmtScope, err = rcmgr.OpenConnectionNoIP(dir, usefd, endpoint)
+	} else {
```
Comment (lidel) [inline core/node/libp2p/rcmgr_logging.go:127]: Shouldn't we do this be as-is and do this assert in  in `func (n *loggingResourceManager) OpenConnection(dir network.Direction, usefd bool, remote ma.Multiaddr)` instead?

Comment (hacdias) [reply core/node/libp2p/rcmgr_logging.go:127]: I don't think so. Otherwise all normal `OpenConnection` will end up in `OpenConnectionNoIP`.

Comment (hacdias) [reply core/node/libp2p/rcmgr_logging.go:127]: And this fixes the test (locally).

File: core/node/libp2p/rcmgr.go
Comment (lidel) [inline core/node/libp2p/rcmgr.go:1]: @hacdias if it helps, sharness `./t0061-daemon-opts.sh -v` seems to fail ([`t0061-daemon-opts` log here](https://custom-github-runners-v0-interplanetary-shipyard.s3.amazonaws.com/multi-linux-x64-4xlarge-action-runner/ipfs/kubo/9179757340/3/sharness.html#t0061-daemon-opts)) with  `missing dependencies for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): missing type: peer.ID`

```
2024-05-21T22:08:17.830Z	ERROR	p2pnode	libp2p/sec.go:14	Your IPFS node has been configured to run WITHOUT ENCRYPTED CONNECTIONS.
		You will not be able to connect to any nodes configured to use encrypted connections
2024-05-21T22:08:17.842Z	ERROR	core	core/builder.go:158	constructing the node: could not build arguments for function "github.com/ipfs/kubo/core/node".PeerWith.func1 (github.com/ipfs/kubo/core/node/peering.go:30): failed to build *peering.PeeringService: could not build arguments for function "github.com/ipfs/kubo/core/node".Peering (github.com/ipfs/kubo/core/node/peering.go:14): failed to build host.Host: received non-nil error from function "github.com/ipfs/kubo/core/node/libp2p".Host (github.com/ipfs/kubo/core/node/libp2p/host.go:41): could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): could not build value group transport.Transport[group="transport"]: could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): failed to build transport.Upgrader: could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): failed to build []sec.SecureTransport[name="security"]: missing dependencies for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): missing type: peer.ID
Error: constructing the node (see log for full detail): could not build arguments for function "github.com/ipfs/kubo/core/node".PeerWith.func1 (github.com/ipfs/kubo/core/node/peering.go:30): failed to build *peering.PeeringService: could not build arguments for function "github.com/ipfs/kubo/core/node".Peering (github.com/ipfs/kubo/core/node/peering.go:14): failed to build host.Host: received non-nil error from function "github.com/ipfs/kubo/core/node/libp2p".Host (github.com/ipfs/kubo/core/node/libp2p/host.go:41): could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): could not build value group transport.Transport[group="transport"]: could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): failed to build transport.Upgrader: could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): failed to build []sec.SecureTransport[name="security"]: missing dependencies for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): missing type: peer.ID
```


(I was able to reproduce it locally too)

Comment (hacdias) [reply core/node/libp2p/rcmgr.go:1]: Me too, something is failing when creating a new libp2p host...

Comment (hacdias) [reply core/node/libp2p/rcmgr.go:1]: We’re getting a dependency injection error when calling `libp2p.New` with `libp2p.NoSecurity` (only used in tests AFAIK):

```
could not build arguments for function "reflect".makeFuncStub (reflect/asm_arm64.s:29): could not build value group transport.Transport[group="transport"]: could not build arguments for function "reflect".makeFuncStub (reflect/asm_arm64.s:29): failed to build transport.Upgrader: could not build arguments for function "reflect".makeFuncStub (reflect/asm_arm64.s:29): failed to build []sec.SecureTransport[name="security"]: missing dependencies for function "reflect".makeFuncStub (reflect/asm_arm64.s:29): missing type: peer.ID
```

**Pull request merged**